src/composables/useNavigationGuards.ts

```diff
@@ -1,18 +1,98 @@
 import NProgress from 'nprogress';
 import { useRouter } from 'vue-router';
 
+import { Network } from '@balancer-labs/sdk';
+import config from '@/lib/config';
+
 import { useSidebar } from './useSidebar';
 import useVeBal from './useVeBAL';
+import { networkFor, networkFromSlug, appUrl } from '@/composables/useNetwork';
 
 // Progress bar config
 NProgress.configure({ showSpinner: false });
 let delayedStartProgressBar;
 
+// Get top level domain excluding 'beta' or 'staging'
+export function getTopLevelDomain(url: string) {
+  const betaEnv =
+    url.split('.')[0] === 'beta' || url.split('.')[0] === 'staging';
+  return betaEnv ? url.split('.')[1] : url.split('.')[0];
+}
+
+// old format subdomain redirect - e.g. "https://polygon.balancer.fi/"
+// if conflicting formats - use the url - e.g. "https://polygon.balancer.fi/#/arbitrum" -> arbitrum
+export function getOldFormatRedirectUrl(
+  networkFromSubdomain: Network,
+  networkSlug: string | undefined,
+  fullPath: string
+) {
+  const networkFromUrl = networkFromSlug(networkSlug ?? '');
+  localStorage.setItem(
+    'networkId',
+    (networkFromUrl || networkFromSubdomain).toString()
+  );
+  return networkFromUrl
+    ? `${appUrl()}${fullPath}`
+    : `${appUrl()}/${config[networkFromSubdomain].slug}${fullPath}`;
+}
```

Comment:
Again, not super clear from the code what's going on here. I feel like the inputs here are over-complicating things. We should have a single function which takes the current URL and outputs the redirect URL if required

```suggestion
/**
 * Given current URL returns redirect URL, if no redirect required returns input URL.
 *
 * @param {string} url - Current URL.
 * @returns {string} URL to redirect too or current URL if no redirect required.
 */
export function getSubdomainNetworkRedirect(url: string): string {
  const subdomainNetwork = getSubdomainNetwork(url);
  if (subdomainNetwork) {
    // Construct redirect URL
  } else {
    return url;
  }
}
```
With a function like this we can more easily and clearly test; given input URL I expect this output URL.

Reply:
Thanks. How would we get fullPath and networkSlug from a url? By using regex?

I changed the function to this format but it requires more parameters.